### ISSUE TITLE
Consul integration plugin

### ISSUE THREAD

**Issue by derekperkins** [goadesign/goa#251]
Potential DSL plugin
- [Service Definitions](https://www.consul.io/docs/agent/services.html)
- [Health checks](https://www.consul.io/docs/agent/checks.html) Dependent on #230

**Comment by joeblew99** [comment]
This sounds interesting..
My thoughts on what this woudl need to do are:
1. Describe the many microservices, so that you can compound them
2. Integrate with Configuration data that it provides for al my microservices, so that when there many instances of the same microservices, they all automatically get the same config
3. As per 2, but them allow things above know the new endpoints. For example a Rervse Proxy, needs to know thow the Endpoints of the child microservices it is proxying.

In order to not reinvent the wheel we could integrate the spf13 packages into the code gen, that work well with concul:
https://github.com/spf13/viper
https://github.com/spf13/cobra (maybe not needed)

also there is https://github.com/spf13/afero
this abstracts out file systems.

**Comment by raphael** [comment]
Closing this old issue because no activity, feel free to open a new issue.

**Comment by raphael** [comment]
Closing this old issue because no activity, feel free to open a new issue.
